oauth2_http/javatests/com/google/auth/oauth2/OAuth2CredentialsTest.java

```diff
@@ -873,6 +871,44 @@ public void serialize() throws IOException, ClassNotFoundException {
     assertSame(deserializedCredentials.clock, Clock.SYSTEM);
   }
 
+  @Test
+  public void UpdateTokenValueBeforeWake()
+      throws IOException, ExecutionException, InterruptedException {
+    // final String accessToken = "1/MkSJoj1xsli0AccessToken_NKPY2";
+    // MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    // transportFactory.transport.addClient(CLIENT_ID, CLIENT_SECRET);
+    // transportFactory.transport.addRefreshToken(REFRESH_TOKEN, accessToken);
+    // OAuth2Credentials userCredentials =
+    // UserCredentials.newBuilder()
+    // .setClientId(CLIENT_ID)
+    // .setClientSecret(CLIENT_SECRET)
+    // .setRefreshToken(REFRESH_TOKEN)
+    // .setHttpTransportFactory(transportFactory)
+    // .build();
+    OAuth2Credentials creds =
+        new OAuth2Credentials() {
+          @Override
+          public void refresh() {
```

Comment:
couple things that might help: 

1. you can have a class variable with a default runnable set, visible for testing. 
2. you don't override the refresh(), you override the refreshAccessToken and override the runnable. And you call the refresh.